rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/monitoring/views/compact/NodeRemover.java

```diff
@@ -0,0 +1,281 @@
+            return Optional.of(index);
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    /**
+     * Tries to find host with the same id (hostName) as given
+     * @param host
+     * @return index of host if found
+     */
+    public Optional<Integer> find(Host host) {
+        init(host);
+        if(findNodeSource() && findHost()){
+            return Optional.of(index);
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    /**
+     * Tries to find node with the same id (nodeUrl) as given
+     * @param node
+     * @return index of node if found
+     */
+    public Optional<Integer> find(Node node){
+        init(node);
+        if(findNodeSource() &&
+                ((node.isDeployingNode() && findDeploying())
+                        || (findHost() && findNode()))){
+            return Optional.of(index);
+        } else {
+            return Optional.empty();
+        }
+    }
+
+
+    /**
+     * Tries to find and remove node source with the same id (sourceName) as given
+     * @param nodeSource
+     */
+    void findAndRemove(NodeSource nodeSource){
+        init(nodeSource);
+        if(findNodeSource()){
+            hierarchyNodeSourceIterator.remove();
+            compactFlowPanel.remove(index);
+        }
+    }
+
+    /**
+     * Tries to find and remove node with the same id (nodeUrl) as given
+     * @param node
+     */
+    public void findAndRemove(Node node) {
+        init(node);
+        if(findNodeSource()){
+            if(node.isDeployingNode() && findDeploying()){
+                removeDeployingNode();
+            } else {
+                if(findHost() && findNode()){
+                    removeNode();
+
+                    // remove dangling host
+                    if (hierarchyHost.isDangling()) {
+                        removeHost();
+                    }
+
+                    return;
+                }
+            }
+        }
+    }
+
+
+    protected void removeHost() {
+        hierarchyHostIterator.remove();
+        hierarchyNodeSource.decrementTiles();
+        compactFlowPanel.remove(index - 1);
+    }
+
+    protected void removeNode() {
+        nodeIterator.remove();
+        hierarchyNodeSource.decrementTiles();
+        hierarchyHost.decrementTiles();
+        compactFlowPanel.remove(index);
+    }
+
+    private void removeDeployingNode() {
+        nodeIterator.remove();
+        hierarchyNodeSource.decrementTiles();
+        compactFlowPanel.remove(index);
+    }
+
+    /**
+     * Tries to find node source, if find, it sets fields dedicated to node source
+     * and make index point to this node source.
+     * @return true if node source was found
+     */
+    boolean findNodeSource(){
+        final Iterator<HierarchyNodeSource> nsIterator = compactFlowPanel.getModel().iterator();
+        while (nsIterator.hasNext()) {
+            final HierarchyNodeSource hierarchyNodeSource = nsIterator.next();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "hierarchyNodeSource" which hides the field declared at line 44. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)